internal/proto/controller/api/resources/scopes/v1/scope.proto

```diff
@@ -85,3 +85,58 @@ message Scope {
   // Output only. The authorized actions for the scope's collections.
   map<string, google.protobuf.ListValue> authorized_collection_actions = 310 [json_name = "authorized_collection_actions"];
 }
+
+// KeyVersion describes a specific version of a key and holds the actual key material
+message KeyVersion {
+  // The ID of the key version.
+  string id = 10; // @gotags: `class:"public"`
+
+  // The iteration of the Key that this version represents.
+  uint32 version = 20; // @gotags: `class:"public"`
+
+  // When this version was created.
+  google.protobuf.Timestamp created_time = 30 [json_name = "created_time"]; // @gotags: `class:"public"`
+}
+
+// Key contains all fields related to a Key in a Scope.
+message Key {
+  // The ID of the Key.
+  string id = 10; // @gotags: `class:"public"`
+
+  // Scope information for this resource.
+  ScopeInfo scope = 20;
+
+  // The purpose of the Key.
+  string purpose = 30; // @gotags: `class:"public"`
+
+  // The time this Key was created.
+  google.protobuf.Timestamp created_time = 40 [json_name = "created_time"]; // @gotags: `class:"public"`
+
+  // The type of the Key.
+  string type = 50; // @gotags: `class:"public"`
+
+  // The versions of the key.
+  repeated KeyVersion versions = 60;
+}
+
+// KeyVersionDestructionJob holds information about a pending key version destruction job.
+message KeyVersionDestructionJob {
+  // The ID of the Key version this job relates to.
+  string key_version_id = 10; // @gotags: `class:"public"`
+
+  // Scope information for this resource.
+  ScopeInfo scope = 20;
+
+  // The current status of the key version destruction job. One of "pending", "running" or "completed".
+  string status = 30; // @gotags: `class:"public"`
```

Comment:
We discussed using an enum, but apparently we never used an enum on the REST API (maybe we don't like the values it generates?) so we didn't think this was a case for introducing one. I'm happy to consider it, but it might be something we'd need to get @talanknight's thoughts on first.

Reply:
Looking over this again... I'm fine with leaving this as-is, especially given the constrained values coming from the view.